kong/target_service_test.go

```diff
@@ -100,13 +100,27 @@ func TestTargetsUpdate(T *testing.T) {
 	createdTarget, err = client.Targets.Create(defaultCtx,
 		fixtureUpstream.ID, &Target{
 			ID:     &targetID,
+			Weight: Int(1),
 			Target: String("10.0.0.2:80"),
 		})
 	require.NoError(err)
-	assert.NotNil(createdTarget)
+	require.NotNil(createdTarget)
 	assert.Equal(targetID, *createdTarget.ID)
+	assert.Equal(1, *createdTarget.Weight)
 	assert.Equal("10.0.0.2:80", *createdTarget.Target)
 
+	updatedTarget, err := client.Targets.Create(defaultCtx,
+		fixtureUpstream.ID, &Target{
+			ID:     &targetID,
+			Weight: Int(2),
+			Target: String("10.0.0.3:80"),
+		})
+	require.NoError(err)
+	require.NotNil(updatedTarget)
+	assert.Equal(targetID, *updatedTarget.ID)
+	assert.Equal(2, *updatedTarget.Weight)
+	assert.Equal("10.0.0.3:80", *updatedTarget.Target)
+
 	err = client.Upstreams.Delete(defaultCtx, fixtureUpstream.ID)
```

Comment:
While we're at it we might as well move this delete above also using `t.Cleanup()`